### PR TITLE
hostapd, wpa_supplicant: Enable 802.11ax (Wi-Fi 6) support

### PR DESCRIPTION
**Pull request by dblsaiko**
###### Description of changes

Compile in IEEE 802.11ax (Wi-Fi 6) support. I use this to host an access point on my router.

The functionality is disabled by default, and needs `ieee80211ax=1` set in the hostapd configuration to enable.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by oddlama** [comment]
This is already covered by #222536. We can theoretically merge this small change first if we don't want to go forward with the big PR in the near future to at least enable WiFi 6 for users.

For completeness I'd also set `CONFIG_IEEE80211AX=y` for `wpa_supplicant` to allow connecting.

**Comment by 2xsaiko** [comment]
> This is already covered by #222536. We can theoretically merge this small change first if we don't want to go forward with the big PR in the near future to at least enable WiFi 6 for users.

I'll keep this for now, partially since I also hope to see RFC 42 in #222536 after all (see my comment) :)

> For completeness I'd also set `CONFIG_IEEE80211AX=y` for `wpa_supplicant` to allow connecting.

Good point. Done!

File: pkgs/os-specific/linux/wpa_supplicant/default.nix [hunk 1]
```diff
@@ -48,6 +48,7 @@ stdenv.mkDerivation rec {
     CONFIG_HS20=y
     CONFIG_HT_OVERRIDES=y
     CONFIG_IEEE80211AC=y
+    CONFIG_IEEE80211AX=y
```
Comment (Tungsten842) [inline pkgs/os-specific/linux/wpa_supplicant/default.nix:51]: This is unneeded, it is already enabled by default [wpa_supplicant](https://w1.fi/cgit/hostap/tree/wpa_supplicant/defconfig#n487)

Comment (2xsaiko) [reply pkgs/os-specific/linux/wpa_supplicant/default.nix:51]: Not in 2.10 https://w1.fi/cgit/hostap/tree/wpa_supplicant/defconfig?h=hostap_2_10

**Review by otavio — APPROVED** [review]

**Pull request merged**